File: army/JavaArmy/src/com/maksklaban/armymax/units/Vampire.java
package com.maksklaban.armymax.units;

import com.maksklaban.armymax.exceptions.*;
import com.maksklaban.armymax.attack.*;

public class Vampire extends Unit {
    public Vampire(String nickName, int damage, int hitPoints, String title) throws CloneNotSupportedException {
        super(nickName, damage, hitPoints, title, true, true, false);
        this.physicalAttack = new AttackVampire();
    }

    public Vampire(String nickName) throws CloneNotSupportedException {
        this(nickName, 30, 180, "Vampire");
    }
}

File: army/Unit/Vampire.cpp
#include "Vampire.h"

Vampire::Vampire(std::string nickName, int damage, int hitPoints, std::string title) : Unit(nickName,damage, hitPoints, title, true, true, false) {
    delete this->physicalAttack;
    this->physicalAttack = new AttackVampire();
}

Vampire::~Vampire() {}
